addons/point_of_sale/models/pos_config.py

```diff
@@ -601,13 +607,10 @@ def assign_payment_journals(self, company):
             cash_journal = self.env['account.journal'].search([('company_id', '=', company.id), ('type', '=', 'cash')], limit=1)
             bank_journal = self.env['account.journal'].search([('company_id', '=', company.id), ('type', '=', 'bank')], limit=1)
             payment_methods = self.env['pos.payment.method']
-            if cash_journal:
+            if cash_journal and  len(self.payment_method_ids.filtered(lambda s: s.journal_id.id in cash_journal.ids)) > 1:
                 payment_methods |= payment_methods.create({
                     'name': _('Cash'),
-                    'journal_id': self.env['account.journal'].create({
-                                    'name': "Cash",
-                                    'type': 'cash',
-                                }).id,
+                    'journal_id': cash_journal.id,
```

Comment:
That is strange with the condition above make sure you do not have a cash payment method associated with this journal (1 cash payment method per journal)